tests/staggered_invert_test.cpp

```diff
@@ -310,32 +327,49 @@ int main(int argc, char **argv)
   case 2: // full parity solution, solving ODD ODD prec system
   case 3: // even parity solution, solving EVEN system
   case 4: // odd parity solution, solving ODD system
-
     if (multishift != 1) {
       printfQuda("Multishift not supported for test %d\n", test_type);
       exit(0);
     }
 
-    for (int k = 0; k < Nsrc; k++) {
-      quda::spinorNoise(*in[k], *rng, QUDA_NOISE_UNIFORM);
-      if (inv_deflate) eig_param.preserve_deflation = k < Nsrc - 1 ? QUDA_BOOLEAN_TRUE : QUDA_BOOLEAN_FALSE;
-      invertQuda(out->V(), in[k]->V(), &inv_param);
+    for (int k = 0; k < Nsrc; k++) { quda::spinorNoise(*in[k], *rng, QUDA_NOISE_UNIFORM); }
+
+    if (!use_split_grid) {
+      for (int k = 0; k < Nsrc; k++) {
+        if (inv_deflate) eig_param.preserve_deflation = k < Nsrc - 1 ? QUDA_BOOLEAN_TRUE : QUDA_BOOLEAN_FALSE;
+        invertQuda(out[k]->V(), in[k]->V(), &inv_param);
+        time[k] = inv_param.secs;
+        gflops[k] = inv_param.gflops / inv_param.secs;
+        iter[k] = inv_param.iter;
+        printfQuda("Done: %i iter / %g secs = %g Gflops\n\n", inv_param.iter, inv_param.secs,
+                   inv_param.gflops / inv_param.secs);
+      }
+    } else {
+      std::vector<void *> _hp_x(Nsrc);
+      std::vector<void *> _hp_b(Nsrc);
+      for (int k = 0; k < Nsrc; k++) {
+        _hp_x[k] = out[k]->V();
+        _hp_b[k] = in[k]->V();
+      }
+      inv_param.num_src = Nsrc;
+      inv_param.num_src_per_sub_partition = Nsrc / num_sub_partition;
+      invertMultiSrcStaggeredQuda(_hp_x.data(), _hp_b.data(), &inv_param, (void *)milc_fatlink, (void *)milc_longlink,
+                                  &gauge_param);
+      // TODO: collect the gflops, etc on all sub-partitions.
+    }
 
-      time[k] = inv_param.secs;
-      gflops[k] = inv_param.gflops / inv_param.secs;
-      iter[k] = inv_param.iter;
-      printfQuda("Done: %i iter / %g secs = %g Gflops\n\n", inv_param.iter, inv_param.secs,
-                 inv_param.gflops / inv_param.secs);
+    for (int k = 0; k < Nsrc; k++) {
       if (verify_results)
-        verifyStaggeredInversion(tmp, ref, in[k], out, mass, qdp_fatlink, qdp_longlink, (void **)cpuFat->Ghost(),
+        verifyStaggeredInversion(tmp, ref, in[k], out[k], mass, qdp_fatlink, qdp_longlink, (void **)cpuFat->Ghost(),
                                  (void **)cpuLong->Ghost(), gauge_param, inv_param, 0);
     }
-
     break;
 
   case 5: // multi mass CG, even parity solution, solving EVEN system
   case 6: // multi mass CG, odd parity solution, solving ODD system
 
+    if (use_split_grid) { errorQuda("Multishift currently doesn't support split grid.\n"); }
```

Comment:
Any particular reason?

Reply:
I am like, tired, of adding a new interface called `invertMultiShiftMultiSrcQuda`?

Reply:
I mean, you can do it now, or file an issue and do it in the future. In either case we should have uniform split grid support, at least on the fine level.

Reply:
#1091.